File: Structural/Facade/JavaScript/projector.js
// projector.js

class Projector {
  turnOn() {
    console.log("Projector is ON.");
  }

  turnOff() {
    console.log("Projector is OFF.");
  }

  setInput(inputSource) {
    console.log(`Projector input set to: ${inputSource}`);
  }
}

module.exports = Projector;

File: Structural/Facade/Cpp/Projector.cpp
// Projector.cpp

#include "Projector.h"

void Projector::turnOn() {
  std::cout << "Projector is ON." << std::endl;
}

void Projector::turnOff() {
  std::cout << "Projector is OFF." << std::endl;
}

void Projector::setInput(const std::string& input) {
  std::cout << "Projector input set to: " << input << std::endl;
}
